--- AMD/Include/amd.h.orig	2014-11-21 18:37:31.104213554 +0100
+++ AMD/Include/amd.h	2014-11-21 18:46:11.707525119 +0100
@@ -35,16 +35,16 @@
 #ifndef AMD_H
 #define AMD_H
 
-/* make it easy for C++ programs to include AMD */
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 /* get the definition of size_t: */
 #include <stddef.h>
 
 #include <SuiteSparse_config.h>
 
+/* make it easy for C++ programs to include AMD */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 int amd_order                  /* returns AMD_OK, AMD_OK_BUT_JUMBLED,
                                 * AMD_INVALID, or AMD_OUT_OF_MEMORY */
 (
