--- AMD/Demo/Makefile.orig	2012-05-27 01:13:34.000000000 +0200
+++ AMD/Demo/Makefile	2012-06-27 15:15:08.539047966 +0200
@@ -4,11 +4,11 @@
 
 default: amd_simple amd_demo amd_demo2 amd_l_demo
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
-C = $(CC) $(CF) -I../Include -I../../SuiteSparse_config
+C = $(CC) $(CF) -I../Include
 
-INC = ../Include/amd.h ../../SuiteSparse_config/SuiteSparse_config.h
+INC = ../Include/amd.h
 
 library:
 	( cd ../Lib ; $(MAKE) )
--- AMD/Doc/Makefile.orig	2012-05-28 02:24:04.000000000 +0200
+++ AMD/Doc/Makefile	2012-06-27 15:15:47.109046348 +0200
@@ -4,7 +4,7 @@
 
 default: dist
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 #------------------------------------------------------------------------------
 # Remove all but the files in the original distribution
--- AMD/Include/amd.h.orig	2012-05-29 18:42:50.000000000 +0200
+++ AMD/Include/amd.h	2012-06-27 15:12:34.369054437 +0200
@@ -43,7 +43,7 @@
 /* get the definition of size_t: */
 #include <stddef.h>
 
-#include "SuiteSparse_config.h"
+#include <SuiteSparse_config.h>
 
 int amd_order                  /* returns AMD_OK, AMD_OK_BUT_JUMBLED,
                                 * AMD_INVALID, or AMD_OUT_OF_MEMORY */
--- AMD/Lib/Makefile.orig	2012-05-27 01:19:39.000000000 +0200
+++ AMD/Lib/Makefile	2012-06-27 15:16:07.689045486 +0200
@@ -8,9 +8,9 @@
 
 default: everything
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
-C = $(CC) $(CF) -I../Include -I../../SuiteSparse_config
+C = $(CC) $(CF) -I../Include
 
 everything:
 	$(C) -DDINT -c ../Source/amd_aat.c -o amd_i_aat.o
--- AMD/Makefile.orig	2012-05-28 22:30:07.000000000 +0200
+++ AMD/Makefile	2012-06-27 15:16:34.285711036 +0200
@@ -6,7 +6,7 @@
 
 default: all
 
-include ../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 demos: all
 
--- AMD/Lib/GNUmakefile.orig	2012-05-29 23:56:26.000000000 +0200
+++ AMD/Lib/GNUmakefile	2012-06-27 15:20:21.892368141 +0200
@@ -4,9 +4,9 @@
 
 default: libamd.a
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
-C = $(CC) $(CF) -I../Include -I../../SuiteSparse_config
+C = $(CC) $(CF) -I../Include
 
 #-------------------------------------------------------------------------------
 # source files
@@ -15,9 +15,7 @@
 AMD = amd_aat amd_1 amd_2 amd_dump amd_postorder amd_post_tree amd_defaults \
 	amd_order amd_control amd_info amd_valid amd_preprocess
 
-CONFIG = ../../SuiteSparse_config/SuiteSparse_config.h
-
-INC = ../Include/amd.h ../Include/amd_internal.h $(CONFIG)
+INC = ../Include/amd.h ../Include/amd_internal.h
 
 #-------------------------------------------------------------------------------
 # object files for each version
